crates/wash-cli/src/cmd/wit/deps.rs

```diff
@@ -0,0 +1,321 @@
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+
+use anyhow::{bail, Context, Result};
+use async_compression::tokio::bufread::GzipDecoder;
+use clap::Args;
+use futures::TryStreamExt as _;
+use reqwest::Url;
+use walkdir::WalkDir;
+use wash_lib::build::{load_lock_file, monkey_patch_fetch_logging};
+use wash_lib::cli::{CommandOutput, CommonPackageArgs};
+use wash_lib::common::clone_git_repo;
+use wash_lib::parser::{
+    load_config, CommonConfig, ProjectConfig, RegistryConfig, RegistryPullConfig,
+    RegistryPullSource, RegistryPullSourceOverride,
+};
+use wasm_pkg_client::PackageRef;
+use wasm_pkg_core::config::Override;
+use wasm_pkg_core::wit::OutputType;
+use wasmcloud_core::parse_wit_package_name;
+
+/// Arguments to `wash wit deps`
+#[derive(Debug, Args, Clone)]
+pub struct DepsArgs {
+    /// The directory containing the WIT files to fetch dependencies for.
+    #[clap(short = 'd', long = "wit-dir", default_value = "wit")]
+    pub dir: PathBuf,
+
+    /// The desired output type of the dependencies. Valid options are "wit" or "wasm" (wasm is the
+    /// WIT package binary format).
+    #[clap(short = 't', long = "type")]
+    pub output_type: Option<OutputType>,
+
+    #[clap(flatten)]
+    pub common: CommonPackageArgs,
+
+    /// Path to the wasmcloud.toml file or parent folder to use for building
+    #[clap(short = 'p', long = "config-path")]
+    config_path: Option<PathBuf>,
+}
+
+/// Invoke `wash wit deps`
+pub async fn invoke(
+    DepsArgs {
+        dir,
+        common,
+        config_path,
+        ..
+    }: DepsArgs,
+) -> anyhow::Result<CommandOutput> {
+    // Load wasmcloud.toml configuration, if present
+    let project_config = match load_config(config_path, Some(true)).await {
+        Ok(v) => Some(v),
+        Err(e) => {
+            eprintln!("failed to load project configuration: {e}");
+            None
+        }
+    };
+
+    // Extract/build configuration for wkg
+    let mut wkg_config = match project_config {
+        Some(ProjectConfig {
+            ref package_config, ..
+        }) => package_config.clone(),
+        None => wasm_pkg_core::config::Config::load().await?,
+    };
+    let mut wkg_config_overrides = wkg_config.overrides.take().unwrap_or_default();
```

Comment:
Nit: Might be worth adding a comment why you're temporarily removing this here

Reply:
Sure -- going to just move this into the helper function!